### PR TITLE
Feat/modularise UI kit

### PR DESCRIPTION
**Pull request by radireddy**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests 

Components modularization

### Description of the Change

Below changes are made 
1. Move all styles, icons, and flags from @ovh-ux/ui-kit.core to @ovh-ux/ui-kit.css package
2. Delete package @ovh-ux/ui-kit.core
3. Replace all references of @ovh-ux/ui-kit.core with @ovh-ux/ui-kit.css
4. Move package @ovh-ux/ui-kit.components from /packages/base to /packages.
5. Build and export each component separately. This enables users to only import components they require.
6. Update docs to use component modularization.

### Benefits

Added support for component modularization. This enables users to only import components they require. This improves page load performance by reducing bundle size.

### CODE REVIEW

File: packages/css/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    ],
+    'oui-icons': [
+      './src/oui-icons.less',
+    ],
+    oui: [
+      './src/index.less',
+    ],
+  },
+  output: {
+    filename: '[name].js',
```
Comment (JeremyDec) [inline packages/css/webpack.config.js:24]: **question**:
Do you need to produce js file output for this package ? I mean, can we have just a css output file here, as that is what the package is supposed to provide ?

**Comment by zakariasahmane** [comment]
Hi Ravi ! 

I see here that we use webpacks to generate chunks of each component. 
That can be heavy to maintain because the webpack config is closely tied to the code. 

Couldn't we use ES6 modules instead? I don't know if it's possible to do so in this context ( and i also don't know how it was made before), but i think that there is an option in the package.json called "module" which is given one js file that contains es modules, and i think that this option tell the bundlers that we want to allow ES Modules and therefore allow the treeshaking for the package.
This way, we can import a specific component without importing the whole library thanks to the treeshaking. 

I am not sure of this being available with angular.js and if it's even possible, it's just an idea thrown out there but i know that's how many modern libraries allow the treeshaking thus making their libs lighter.

**Review by JeremyDec — CHANGES_REQUESTED** [review]

File: packages/components/README.md [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+
+2. You need to add `{componentName}` in your angular module dependencies like:
+
+```js
+angular.module('myAwesomeApp', ['{componentName}']);
+```
+
+Note: Replace {componentName} with actual component like 'oui-button' for button component.
+
+Refer to component README file for more details.
```
Comment (JeremyDec) [inline packages/components/README.md]: **suggestion**
Maybe you could directly show an example with `oui-button`. (For `js` or `html` part)

File: packages/components/README.md [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+import '@ovh-ux/ui-kit.css/dist/css/oui.css';
+import ouiComponent from  '@ovh-ux/ui-kit.components/dist/js/oui-{componentName}.js';
+angular.module('myAwesomeApp', ['ouiComponent']);
+```
+
+Note: Replace {componentName} with actual component like 'button' for button component.
+
+Refer to component README file for more details.
+
+## Documentation
```
Comment (JeremyDec) [inline packages/components/README.md]: You could also mention the Storybook documentation url: https://ovh.github.io/ovh-ui-kit/

File: packages/apps/workshop/README.md [hunk 4]
```diff
@@ -6,6 +6,7 @@ Install and run Storybook in a dev mode
 
 ```sh
 yarn
+yarn build:dependency
```
Comment (marie-j) [inline packages/apps/workshop/README.md:9]: Shouldn't this be integrated in `yarn:start` ? I'm not sure we should be worrying what to start and to build when starting storybook. For instance in `manager` repository we have the `start:dev` command that will handle the build of needed dependencies so we don't have to worry about it

File: packages/components/action-menu/src/index.less [hunk 5]
```diff
@@ -1,4 +0,0 @@
-@import '~@ovh-ux/ui-kit.core/src/index';
-@import '~@ovh-ux/ui-kit.button/src/index';
-@import '~@ovh-ux/ui-kit.dropdown/src/index';
-@import './less/action-menu';
```
Comment (marie-j) [inline packages/components/action-menu/src/index.less:4]: I'm having trouble understanding the removal of the less file. Shouldn't the component still import its style to be standalone ?
`@import './less/action-menu';`

File: packages/apps/workshop/stories/documentation/migrate/from-v4-to-v5.story.mdx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { Meta, Description } from '@storybook/addon-docs/blocks';
+import readme from '@ovh-ux/ui-kit/V4-TO-V5.md';
+
+<Meta title="Documentation/Migrate/From v4.x to v5.x" />
+
+<Description>{readme}</Description>
```
Comment (marie-j) [inline packages/apps/workshop/stories/documentation/migrate/from-v4-to-v5.story.mdx:6]: SHould be completed ?

File: packages/components/webpack.config.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+    ],
+    'oui-slideshow': [
+      './slideshow/src/index.js',
+    ],
+    'oui-header': [
+      './header/src/index.js',
+    ],
+    'oui-dual-list': [
+      './dual-list/src/index.js',
+    ],
```
Comment (marie-j) [inline packages/components/webpack.config.js:132]: Not really easy to maintain, if we want to add a component, we have to update the webpack.config

File: packages/css/src/less/components/atoms.less [hunk 8]
```diff
@@ -0,0 +1,11 @@
+@import './button/index';
```
Comment (marie-j) [inline packages/css/src/less/components/atoms.less:1]: Why some files are located in `./less` and some directly in `./{component}`. Shouldn't we be consistent ?

File: packages/css/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    library: 'oui',
+    libraryTarget: 'umd',
+    path: path.resolve('.', 'dist', 'js'),
+    // libraryTarget: 'umd',
```
Comment (marie-j) [inline packages/css/webpack.config.js:28]:
```suggestion

```

File: packages/css/webpack.config.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    ],
+    oui: [
+      './src/index.less',
+    ],
+  },
+  output: {
+    filename: '[name].js',
+    library: 'oui',
+    libraryTarget: 'umd',
+    path: path.resolve('.', 'dist', 'js'),
```
Comment (marie-j) [inline packages/css/webpack.config.js:27]: Storing CSS dist in `/js` ?

File: packages/libs/ui-kit/README.md [hunk 11]
```diff
@@ -54,9 +55,25 @@ yarn add @ovh-ux/ui-kit
 ```js
 angular.module('myAwesomeApp', ['oui']);
 ```
+### Import only components required
+1. In your `index.html`, you need to load `oui.css` and `<component-name>.js`:
+
+```html
+<link rel="stylesheet" href="path/to/@ovh-ux/ui-kit/dist/css/oui.css">
```
Comment (marie-j) [inline packages/libs/ui-kit/README.md:62]: So this means that we need to bring the whole style for only one component ? Shouldn't the component bring its own style ?

File: packages/apps/workshop/stories/design-system/components/action-menu/action-menu-webcomponent.stories.js [hunk 12]
```diff
@@ -2,14 +2,14 @@ import { action } from '@storybook/addon-actions';
 import { boolean } from '@storybook/addon-knobs';
 import { forModule } from 'storybook-addon-angularjs';
 
-import ActionMenu from '@ovh-ux/ui-kit.action-menu';
+import '@ovh-ux/ui-kit/dist/js/oui-action-menu';
```
Comment (JeremyDec) [inline packages/apps/workshop/stories/design-system/components/action-menu/action-menu-webcomponent.stories.js:5]: This is breaking the functioning of workshop application.
As you can see below with the `README.md` import, you can import without precising the path of the main file.

```suggestion
import '@ovh-ux/ui-kit.action-menu';
```

File: packages/apps/workshop/stories/design-system/components/button/button-webcomponent-primary.stories.js [hunk 13]
```diff
@@ -1,14 +1,14 @@
 import { boolean } from '@storybook/addon-knobs';
 import { forModule } from 'storybook-addon-angularjs';
 
-import Button from '@ovh-ux/ui-kit.button';
+import ouiButton from '@ovh-ux/ui-kit/dist/js/oui-button';
```
Comment (JeremyDec) [inline packages/apps/workshop/stories/design-system/components/button/button-webcomponent-primary.stories.js:4]: To keep consistency with other stories
```suggestion
import '@ovh-ux/ui-kit/dist/js/oui-button';
```

File: packages/apps/workshop/stories/design-system/components/action-menu/action-menu-webcomponent.stories.js (hunk 12 above)
Comment (JeremyDec) [inline packages/apps/workshop/stories/design-system/components/action-menu/action-menu-webcomponent.stories.js:5]: This is breaking the functioning of workshop application.
As you can see below with the `README.md` import, you can import without precising the path of the main file.

```suggestion
import '@ovh-ux/ui-kit.action-menu';
```

**Review by JeremyDec — CHANGES_REQUESTED** [review]

**Pull request closed without merging**